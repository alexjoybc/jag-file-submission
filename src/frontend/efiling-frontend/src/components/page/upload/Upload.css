.dropzone-outer-box {
  border: 1px dashed #707070;
  height: 250px;
  display: flex;
  justify-content: center;
}

.dropzone-inner-box {
  display: flex;
}

.dropzone-inner-box span {
  align-self: center;
}

.dropzone-inner-box div {
  text-align: center;
}

.file-href {
  cursor: pointer;
  color: #1a5a96;
  text-decoration: underline;
}
