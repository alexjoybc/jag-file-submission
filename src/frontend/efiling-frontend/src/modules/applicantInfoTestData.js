const applicantInfo = {
  bceID: "bobross42",
  firstName: "Bob",
  lastName: "Ross",
  emailAddress: "bob.ross@example.com"
};

export function getApplicantInfo() {
  return applicantInfo;
}
