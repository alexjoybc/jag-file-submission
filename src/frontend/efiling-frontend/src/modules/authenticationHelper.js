const jwt = require("jsonwebtoken");

export function getJWTData() {
  const token = localStorage.getItem("jwt");
  if (!token) return false;

  // get the decoded payload and header
  const decoded = jwt.decode(token, { complete: true });
  return decoded.payload;
}

export function generateJWTToken(payload) {
  return jwt.sign(payload, "something");
}
