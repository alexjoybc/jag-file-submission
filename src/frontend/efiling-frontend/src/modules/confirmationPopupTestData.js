/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn"
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width"
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width"
};

const handleClose = () => {};
const handleShow = () => {};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: true,
    handleShow,
    handleClose,
    handleConfirm: () => {},
    title: "title",
    body
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
