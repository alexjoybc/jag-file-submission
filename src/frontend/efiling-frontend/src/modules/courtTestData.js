const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
};

export function getCourtData() {
  return court;
}
