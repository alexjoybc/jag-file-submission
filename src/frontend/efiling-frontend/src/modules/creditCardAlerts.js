import React from "react";
import { MdCreditCard } from "react-icons/md";
import { Alert } from "shared-components";

const existingCreditCard = () => {
  return (
    <Alert
      icon={<MdCreditCard size={32} />}
      type="success"
      styling="success-background no-padding-bottom"
      element={
        <p>
          <span>
            You have a valid Credit Card registered with your CSO account.
          </span>
          <br />
          <a
            href="https://justice.gov.bc.ca/cso/about/index.do"
            target="_blank"
            rel="noopener noreferrer"
          >
            Register a new Credit Card.
          </a>
        </p>
      }
    />
  );
};

const noCreditCard = () => {
  return (
    <Alert
      icon={<MdCreditCard size={32} />}
      type="error"
      styling="error-background no-padding-bottom"
      element={
        <p>
          <span>
            You do not have a valid Credit Card registered with your CSO
            account.
          </span>
          <br />
          <a
            href="https://justice.gov.bc.ca/cso/about/index.do"
            target="_blank"
            rel="noopener noreferrer"
          >
            Register a Credit Card now
          </a>
          &nbsp;to continue.
        </p>
      }
    />
  );
};

export function getCreditCardAlerts() {
  const existingCreditCardData = existingCreditCard();
  const noCreditCardData = noCreditCard();

  return {
    existingCreditCard: existingCreditCardData,
    noCreditCard: noCreditCardData,
  };
}
