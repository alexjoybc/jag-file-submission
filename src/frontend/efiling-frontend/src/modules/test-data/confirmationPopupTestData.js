/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "normal-blue btn consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "normal-white btn consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return { modal, mainButton, confirmButton, cancelButton };
}
