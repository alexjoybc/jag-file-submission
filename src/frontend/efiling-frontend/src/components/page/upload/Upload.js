import React, { useState } from "react";
import PropTypes from "prop-types";
import { useDropzone } from "react-dropzone";
import { Sidecard, Button } from "shared-components";
import { getSidecardData } from "../../../modules/sidecardData";
import { propTypes } from "../../../types/propTypes";

import "./Upload.css";
import PackageConfirmation from "../package-confirmation/PackageConfirmation";

export default function Upload({
  upload: { confirmationPopup, submissionId }
}) {
  const amendmentsSidecard = getSidecardData().amendments;
  const supremeCourtSchedulingSidecard = getSidecardData()
    .supremeCourtScheduling;
  const { acceptedFiles, getRootProps, getInputProps } = useDropzone();
  const [showPackageConfirmation, setShowPackageConfirmation] = useState(false);

  if (showPackageConfirmation) {
    return (
      <PackageConfirmation
        packageConfirmation={{ confirmationPopup, submissionId }}
        csoAccountStatus={{ isNew: false }}
      />
    );
  }

  const files = acceptedFiles.map(file => (
    <li key={file.path}>
      {file.path} - {file.size} bytes
    </li>
  ));

  return (
    <div className="page">
      <div className="content col-md-8">
        <h1>Document Upload</h1>
        <div {...getRootProps({ className: "dropzone-outer-box" })}>
          <div className="dropzone-inner-box">
            <input {...getInputProps()} />
            <span>
              <h2>
                Drag and drop or&nbsp;
                <span className="file-href">choose documents</span>
              </h2>
              <div>
                <span>to upload additional documents to your package</span>
                <br />
                <span>(max 10mb per document)</span>
              </div>
            </span>
          </div>
        </div>
        <br />
        {acceptedFiles.length > 0 && (
          <>
            <h2>Select Description for each uploaded document to continue</h2>
          </>
        )}
        <section className="buttons pt-2">
          <Button
            label="Cancel Upload"
            onClick={() => setShowPackageConfirmation(true)}
            styling="normal-white btn"
          />
          <Button
            label="Continue"
            onClick={() => console.log("on continue")}
            styling="normal-blue btn"
          />
        </section>
      </div>

      <div className="sidecard">
        <Sidecard sideCard={amendmentsSidecard} />
        <Sidecard sideCard={supremeCourtSchedulingSidecard} />
      </div>
    </div>
  );
}

Upload.propTypes = {
  upload: PropTypes.shape({
    confirmationPopup: propTypes.confirmationPopup,
    submissionId: PropTypes.string.isRequired
  }).isRequired
};
