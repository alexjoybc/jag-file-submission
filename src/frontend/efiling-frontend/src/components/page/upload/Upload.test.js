import React from "react";
import { render } from "@testing-library/react";
import { getTestData } from "../../../modules/confirmationPopupTestData";

import Upload from "./Upload";

describe("Upload Component", () => {
  const confirmationPopup = getTestData();
  const submissionId = "abc123";
  const upload = {
    confirmationPopup,
    submissionId
  };

  test("Matches the snapshot", () => {
    const { asFragment } = render(<Upload upload={upload} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
