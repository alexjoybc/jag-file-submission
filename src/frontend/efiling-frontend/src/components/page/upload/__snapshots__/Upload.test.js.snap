exports[`Upload Component successful document upload works as expected 1`] = `<DocumentFragment />`;